src/common/components/header/index.js

```diff
@@ -7,7 +7,7 @@ import styles from './header.css';
 
 const wordmark = 'https://assets.ubuntu.com/v1/d45097a4-snapcraft.io-logotype.svg';
 
-class Header extends Component {
+export class Header extends Component {
```

Comment:
I think having two separate public interfaces for components is a mistake. It means we're not testing the same thing we're integrating, and it makes it easy to use the wrong interface in the wrong place. Any other approach is better than this, in my opinion.

I don't see the issue of passing in some props in JSX. It does delegate ownership of data up the tree, but in many cases this is what we want. The app container should say **"show a header component containing a logged-in navigation item"**. This allows for cleaner separation of presentational and container components and makes it easier to test.